safenode/src/storage/errors.rs

```diff
@@ -23,16 +26,47 @@ pub enum Error {
     /// Chunk not found.
     #[error("Chunk not found: {0:?}")]
     ChunkNotFound(XorName),
+    /// Register not found.
+    #[error("Register not found: {0:?}")]
+    RegisterNotFound(RegisterAddress),
+    /// NetworkData error.
+    #[error("Network data error:: {0}")]
+    NetworkData(#[from] ProtocolError),
+    /// Data authority provided is invalid.
+    #[error("Provided PublicKey could not validate signature {0:?}")]
+    InvalidSignature(PublicKey),
+    /// Bincode error.
+    #[error("Bincode error:: {0}")]
+    Bincode(#[from] bincode::Error),
     /// I/O error.
     #[error("I/O error: {0}")]
     Io(#[from] io::Error),
     /// Hex decoding error.
     #[error("Hex decoding error:: {0}")]
     HexDecoding(#[from] hex::FromHexError),
-    /// No filename found
-    #[error("Path contains no file name: {0}")]
-    NoFilename(PathBuf),
-    /// Invalid filename
-    #[error("Invalid chunk filename: {0}")]
-    InvalidFilename(PathBuf),
+    /// Register command/op destination adddress mistmatch
+    #[error(
+        "Register command destination address ({cmd_dst_addr:?}) \
+        doesn't match stored Register address: {reg_addr:?}"
+    )]
+    RegisterAddrMismatch {
+        cmd_dst_addr: RegisterAddress,
+        reg_addr: RegisterAddress,
+    },
+}
+
+// Convert storage error to messaging error message for sending over the network.
+impl From<Error> for ProtocolError {
```

Comment:
Now that we don't have these in separate crate, where messaging error was in sn_interface and we had the reversed dependency there so that sn_node knew about interface but not the other way around, we could skip this stringifying of errors.

I've always thought it is less than optimal. And now we could actually have these enum variants directly held in a higher level error.